codes/models/modules/architectures/DVDNet_arch.py

```diff
@@ -0,0 +1,68 @@
+import torch
+import torch.nn as nn
+
+def vertcal_upscale(x, upfield=True):
+
+    n, c, h, w = x.shape
+    h *= 2
+    t = x
+    if upfield:
+        out = torch.cat([t, torch.zeros_like(t)], 3)
+    else:
+        out = torch.cat([torch.zeros_like(t), t], 3)
+    out = torch.reshape(out, (n, c, h, w))

```

Comment:
Here could be safer to reshape to (n, c, -1, w) , the correct shape can be infered